### PR TITLE
daemon: improvements vol.2

### PR DESCRIPTION
**Pull request by ngortheone**
Less trivial changes, refactoring of the code structure

This needs https://github.com/freebsd/freebsd-src/pull/669 to be merged first

### CODE REVIEW

File: usr.sbin/daemon/daemon.c [hunk 1]
```diff
@@ -412,12 +405,23 @@ main(int argc, char *argv[])
...
-		/*
-		 * execvp() failed -- report the error. The child is
-		 * now running, so the exit status doesn't matter.
-		 */
+		/* execvp() failed - report error and exit this process */
 		err(1, "%s", argv[0]);
 	}
+
+	/* else: pid > 0
+         * fork succeeded, this is parent branch, this can only when
```
Comment (igalic) [inline usr.sbin/daemon/daemon.c:413]: this is the parent branch

Comment (igalic) [reply usr.sbin/daemon/daemon.c:413]: this can only happen when

File: usr.sbin/daemon/daemon.c [hunk 2]
```diff
@@ -612,26 +612,26 @@ listen_child(int fd, struct log_params *logpar)
...
+			return true;
 		}
 	}
 	/* Upon EOF, we have to flush what's left of the buffer. */
 	if (bytes_read > 0) {
 		do_output(buf, bytes_read, logpar);
 		bytes_read = 0;
 	}
-	return 0;
+	return false;
```
Comment (igalic) [inline usr.sbin/daemon/daemon.c:643]: (this is for o returns, not just this one)

doesn't our style guide say to use

```c
return (value);
```

?

**Review by igalic — APPROVED** [review]
most of this seems sensible

File: usr.sbin/daemon/daemon.c [hunk 3]
```diff
@@ -413,12 +406,23 @@ main(int argc, char *argv[])
...
 		execvp(argv[0], argv);
-		/*
-		 * execvp() failed -- report the error. The child is
-		 * now running, so the exit status doesn't matter.
-		 */
+		/* execvp() failed - report error and exit this process */
 		err(1, "%s", argv[0]);
 	}
+
+	/* else: pid > 0
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:420]: Comment starts on next line for multiline

File: usr.sbin/daemon/daemon.c [hunk 4]
```diff
@@ -179,6 +179,22 @@ main(int argc, char *argv[])
 	sigemptyset(&mask_term);
 	sigemptyset(&mask_orig);
 
+	/* Supervision mode is enabled if one of the following options are used:
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:182]: Comment starts on next line

File: usr.sbin/daemon/daemon.c [hunk 5]
```diff
@@ -205,15 +222,23 @@ main(int argc, char *argv[])
 			break;
 		case 'o':
 			logparams.output_filename = optarg;
+			/* TODO: setting output filename doesn't have to turn the supervision mode on.
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:225]: Ditto

File: usr.sbin/daemon/daemon.c [hunk 6]
```diff
@@ -418,8 +418,8 @@ main(int argc, char *argv[])
 	}
 
 	/* else: pid > 0
-         * fork succeeded, this is parent branch, this can only when
-	 * supervision_enabled
+         * fork succeeded, this is the parent branch, this can only happen when
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:421]: Let's go ahead and squash this back into the commit that added the comment since that's still new to this PR

**Review by kevans91 — CHANGES_REQUESTED** [review]

File: usr.sbin/daemon/daemon.c (hunk 6 above)
Comment (ngortheone) [reply usr.sbin/daemon/daemon.c:421]: ack

File: usr.sbin/daemon/daemon.c (hunk 3 above)
Comment (ngortheone) [reply usr.sbin/daemon/daemon.c]: fixed

File: usr.sbin/daemon/daemon.c (hunk 5 above)
Comment (ngortheone) [reply usr.sbin/daemon/daemon.c]: fixed

File: usr.sbin/daemon/daemon.c (hunk 6 above)
Comment (ngortheone) [reply usr.sbin/daemon/daemon.c]: done

File: usr.sbin/daemon/daemon.c (hunk 4 above)
Comment (ngortheone) [reply usr.sbin/daemon/daemon.c:182]: fixed

**Review by igalic — APPROVED** [review]
👍can't wait for the actual changes ;)

**Comment by ngortheone** [comment]
coming soon :tm:

**Comment by bsdimp** [comment]
Commotted

**Pull request closed without merging**